datadog-tracer-flare/src/lib.rs

```diff
@@ -62,34 +181,59 @@ impl TryFrom<&str> for LogLevel {
 }
 
 pub type RemoteConfigFile = std::sync::Arc<RawFile<Result<RemoteConfigData, anyhow::Error>>>;
+#[cfg(feature = "listener")]
 pub type Listener = SingleChangesFetcher<RawFileStorage<Result<RemoteConfigData, anyhow::Error>>>;
 
 /// Check the `RemoteConfigFile` and return the action that tracer flare needs
-/// to perform
+/// to perform. This function also updates the `TracerFlareManager` state based on the
+/// received configuration.
 ///
 /// # Arguments
 ///
 /// * `file` - RemoteConfigFile received by the Listener.
+/// * `tracer_flare` - TracerFlareManager object to update with the received configuration.
 ///
 /// # Returns
 ///
 /// * `Ok(ReturnAction)` - If successful.
 /// * `FlareError(msg)` - If something fail.
-pub fn check_remote_config_file(file: RemoteConfigFile) -> Result<ReturnAction, FlareError> {
+pub fn check_remote_config_file(
+    file: RemoteConfigFile,
+    tracer_flare: &mut TracerFlareManager,
+) -> Result<ReturnAction, FlareError> {
     let config = file.contents();
     match config.as_ref() {
         Ok(data) => match data {
             RemoteConfigData::TracerFlareConfig(agent_config) => {
                 if agent_config.name.starts_with("flare-log-level.") {
                     if let Some(log_level) = &agent_config.config.log_level {
+                        if let State::Collecting { log_level: _ } = tracer_flare.state {
+                            // Should we return an error instead if we are trying to launch another
+                            // flare while one is already running ?
```

Comment:
See my [other comment](https://github.com/DataDog/libdatadog/pull/1126/files#r2229246650) about comments that ask questions